src/utils/crypto.ts

```diff
@@ -128,22 +128,43 @@ export async function openCryptobox(
 export async function getAddressFromPublicKey(publicKey: string): Promise<string> {
   await sodium.ready
 
-  let plainPublicKey: string
+  const prefixes = {
+    edpk: {
+      length: 54,
+      prefix: Buffer.from(new Uint8Array([6, 161, 159]))
+    },
+    sppk: {
+      length: 55,
+      prefix: Buffer.from(new Uint8Array([6, 161, 161]))
+    },
+    p2pk: {
+      length: 55,
+      prefix: Buffer.from(new Uint8Array([6, 161, 164]))
+    }
+  }
+
+  let prefix: Buffer | undefined
+  let plainPublicKey: string | undefined
   if (publicKey.length === 64) {
+    prefix = prefixes.edpk.prefix
     plainPublicKey = publicKey
-  } else if (publicKey.startsWith('edpk') && publicKey.length === 54) {
-    const edpkPrefixLength = 4
-    const decoded = bs58check.decode(publicKey)
-
-    plainPublicKey = decoded.slice(edpkPrefixLength, decoded.length).toString('hex')
   } else {
+    Object.entries(prefixes).forEach(([key, value]) => {
+      if (publicKey.startsWith(key) && publicKey.length === value.length) {
+        prefix = value.prefix
+        const decoded = bs58check.decode(publicKey)
+        plainPublicKey = decoded.slice(key.length, decoded.length).toString('hex')
+      }
+    })
```

Comment:
@AndreasGassmann wouldn't be better here to have a for loop, so that as soon as `plainPublicKey` is assigned, we can break and not keep checking for the others?

Reply:
Yes, fixed